### PR TITLE
search frontend: resolve predicate access paths

### PR DESCRIPTION
**Pull request by rvantonder**
Updates frontend predicate scanner to recognize dot access. Adds a function to resolve access paths. This will be used to generate autocomplete suggestions.

None of this is being used yet, just building the functionality I'll hook into.

### CODE REVIEW

File: client/shared/src/search/query/predicates.ts [hunk 1]
```diff
@@ -1,15 +1,53 @@
...
+ * for autocomplete suggestions.
+ */
+export const PREDICATES: Access[] = [
+    {
+        name: 'repo',
+        fields: [
+            {
+                name: 'contains',
+                fields: [
+                    { name: 'file' },
```
Comment (camdencheek) [inline client/shared/src/search/query/predicates.ts:18]: Does this mean the frontend will support `repo:contains.file()`?

File: client/shared/src/search/query/predicates.ts [hunk 2]
```diff
@@ -48,18 +86,23 @@ export const scanBalancedParens = (input: string): string | undefined => {
...
+ * Scans predicate syntax of the form field:foo.bar(parameters) and
  * returns the name and parameters components. It checks that:
  *
  * (1) The (field, name) pair is a recognized predicate.
  * (2) The parameters value is well-balanced.
  */
 export const scanPredicate = (field: string, value: string): Predicate | undefined => {
-    const match = value.match(/^[\da-z]+/i)
-    if (!(match && PREDICATES[field] && PREDICATES[field].some(pred => pred === match[0]))) {
+    const match = value.match(/^[a-z.]+/i)
```
Comment (camdencheek) [inline client/shared/src/search/query/predicates.ts:96]: It looks like the tests cover this, so it doesn't actually seem incorrect, but I'm surprised that this dot doesn't need escaped

**Review by camdencheek — APPROVED** [review]

File: client/shared/src/search/query/predicates.ts (hunk 2 above)
Comment (rvantonder) [reply client/shared/src/search/query/predicates.ts:96]: It's inside character class syntax `[]` doesn't need to be escaped.

Comment (camdencheek) [reply client/shared/src/search/query/predicates.ts:96]: Huh. TIL 👍

File: client/shared/src/search/query/predicates.ts (hunk 1 above)
Comment (rvantonder) [reply client/shared/src/search/query/predicates.ts:18]: Yeah, I should update the scope of RFC and we should support this i the backend (basically just set `f.File` and `f.Content` resp. for `contains.file`, `contains.content` :-) )

**Pull request merged**